src/ray/common/ray_config_def.h

```diff
@@ -699,3 +699,11 @@ RAY_CONFIG(std::string, REDIS_SERVER_NAME, "")
 //  The delay is a random number between the interval. If method equals '*',
 //  it will apply to all methods.
 RAY_CONFIG(std::string, testing_asio_delay_us, "")
+
+/// A feature flag to enable pull based health check.
+/// TODO: Turn it on by default
+RAY_CONFIG(bool, pull_based_healthcheck, false)
+RAY_CONFIG(int64_t, health_check_initial_delay_ms, 1000)
```

Comment:
Feel like it is unnecessary? It'd take 30 seconds until the node is marked as dead anyway?

Reply:
No. 30s is old protocol. This is the new protocol. So here, if a node is dead, it only need to take  3s to mark the node dead.